### PR TITLE
Methods

### LINKED ISSUE #19: Struct methods
Assuming there is a `Foo` struct, this would be the method implementation for "some_method":

```
impl Foo.some_method(x: int) {
    self.bar = x
}
```

To call this method, a familiar syntax is used

```
fn main() {
    let some_foo = new Foo {
        bar: 5
    }

    some_foo.some_method(10)
}
```

### PR DESCRIPTION
**Pull request by garritfra**
Fixes #19 

### Description

Adds struct methods

### ToDo

- [x] Proposed feature/fix is sufficiently tested
- [x] Proposed feature/fix is sufficiently documented
- [x] The "Unreleased" section in the changelog has been updated, if applicable

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -91,14 +100,15 @@ pub enum Expression {
     Int(u32),
     Str(String),
     Bool(bool),
+    Selff,
```
Comment (YerinAlexey) [inline src/ast/mod.rs:103]: What's this?

Comment (garritfra) [reply src/ast/mod.rs:103]: The "self" keyword. `Self` is a reserved keyword in rust.

Comment (YerinAlexey) [reply src/ast/mod.rs:103]: Maybe you should add a comment for it so people won't get confused?

Comment (garritfra) [reply src/ast/mod.rs:103]: Done👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,12 +2,19 @@
...
 ## Unreleased
 
+**Features**
+
+- Struct methods
 - Feature flags for backends (`backend_c`, etc.) have been replaced by `--target` (`-t`) command-line option:
 
   ```sh
   sb -t c run examples/hello_world.sb
   ```
```
Comment (YerinAlexey) [inline CHANGELOG.md:12]: Oops, can you move that entry to Changes section?

**Pull request merged**